src/components/ProgressBar.tsx

```diff
@@ -0,0 +1,99 @@
+        )`;
+    };
+    // derive the percentage of the progress, set the max to 100
+    const percentage = Math.min(Math.round((value / max) * 100), 100);
+    // Function to determine the color of the progress bar based on the percentage
+    const getProgressBarColor = () => {
+        if (percentage < 25) return progressColors.first; // Less than 25% progress
+        if (percentage < 50) return progressColors.second; // Between 25% and 50% progress
+        if (percentage < 75) return progressColors.third; // Between 50% and 75% progress
+        return progressColors.fourth; // More than 75% progress
+    }
+
+    // Define the CircularProgressBar component for circular progress bars
+    const CircularProgressBar = () => {
+        const size = 100; // Size of the SVG element
+        const strokeWidth = 10; // Width of the stroke
+        const radius = (size - strokeWidth) / 2; // Radius of the circle
+        const circumference = radius * 2 * Math.PI; // Circumference of the circle
+        const offset = circumference - (percentage / 100) * circumference; // Offset for the stroke-dashoffset
+
+        return (
+            <div className="relative inline-flex items-center justify-center">
+                <svg width={size} height={size} className="transform -rotate-90">
+                    <circle
+                        className="text-gray-200" // Background circle color
+                        strokeWidth={strokeWidth} // Width of the stroke
+                        stroke="currentColor" // Stroke color
+                        fill="transparent" // Fill color
+                        r={radius} // Radius of the circle
+                        cx={size / 2} // X-coordinate of the center
+                        cy={size / 2} // Y-coordinate of the center
+                    />
+                    <circle
+                        className="transition-all duration-300 ease-in-out" // Transition properties
+                        strokeWidth={strokeWidth} // Width of the stroke
+                        strokeDasharray={circumference} // Dash array for the stroke
+                        strokeDashoffset={offset} // Dash offset for the stroke
+                        strokeLinecap="round" // Line cap style
+                        stroke={getProgressBarColor()} // Stroke color based on progress
+                        fill="transparent" // Fill color
+                        r={radius} // Radius of the circle
+                        cx={size / 2} // X-coordinate of the center
+                        cy={size / 2} // Y-coordinate of the center
+                    />
+                </svg>
+                <span className="absolute text-sm font-bold">{percentage}%</span> {/* Display the percentage */}
+            </div>
+        );
+    }
+
+    return (
+        <>
+            {shape === 'circular' && <CircularProgressBar />} {/* Render circular progress bar if shape is circular */}
```

Comment:
nit - if we thought we might want to add more shapes in the future, I would change this to a switch case.

I usually default to switch cases when making polymorphic components.